pylib/base/exec_utils.py

```diff
@@ -72,24 +74,34 @@ def ExecuteParallel(args, pool_size=0, callback=PicklableCallback()):
     if not pool_size:
       pool_size = max(multiprocessing.cpu_count(), 1)
 
-    # NOTE(stephen): Require use of `fork` for multiprocessing instead of `spawn` since
-    # there are issues with the reuse of some libraries (like Flags) when `spawn` is
-    # used.
-    pool = multiprocessing.get_context('fork').Pool(processes=pool_size)
+    in_queue = Queue()
+    for job in args:
+        in_queue.put(job)
+    out_queue = Queue()
+    def worker(in_queue, out_queue):
+      while (job := in_queue.get()):
+        out_queue.put(callback(job))
+
+    for _ in range(pool_size):
+      Thread(target=worker, args=(in_queue, out_queue)).start()
+
     try:
-      res = pool.map(callback, args)
-      pool.terminate()
-      sys.stdout.flush()
-      return res
+      while True:
```

Comment:
Would this not lock the process when sigterm is raised?

Reply:
I could think that using a daemon thread could be useful but it just works in practice, I stopped it early so many times while testing, no issues around this